File: TwoPointers/344_Reverse_String/go/main.go
func reverseString(s string) string {
	x := []byte(s)

	for i := 0; i < len(s)/2; i++ {
		x[i], x[(len(s)-1)-i] = x[(len(s)-1)-i], x[i]
	}
	return string(x[:])
}

File: TwoPointers/344_Reverse_String/cpp/main.cpp
#include <vector>

using namespace std;

void reverseString(vector<char>& s) {
    int backwardIndex = s.size() - 1;
    for (vector<char>::size_type i=0; i< (s.size()/2); ++i) {
        char temp = s[i];
        s[i] = s[backwardIndex];
        s[backwardIndex] = temp;
        --backwardIndex;
    }
}

int main() {
    vector<char> characters = {'c','o','o'};
    reverseString(characters);
    return 0;
}
